src/routes/login.ts

```diff
@@ -0,0 +1,53 @@
+import { Request, Response } from 'express';
+import { body } from 'express-validator';
+import jwt from 'jsonwebtoken';
+import { User } from '../models';
+import { PasswordManager } from '../util/hash/index';
+
+const loginRouter = require('express').Router();
+
+loginRouter.post(
+  '/api/v1/users/login',
+  [
+    body('username').isLength({ min: 1, max: 20 }).escape(),
+    body('email').isLength({ max: 30 }).isEmail().escape(),
+    body('password').isLength({ min: 2, max: 20 }).escape(),
+  ],
+  async (req: Request, res: Response) => {
+    const reqbody = req.body;
+    // Find User in Database
+    const user = await User.findOne({ username: reqbody.username });
+    const passwordCorrect =
+      user === null
```

Comment:
A good conditional here yet a small tip. 

You could check on top if there's a user and avoid the hassle.

Something like,

```ts
if(!user) return throw new Error("User not found")
```

Reply:
I should've caught that one ! Thanks!